Source/DafnyCore/Triggers/TriggersCollector.cs

```diff
@@ -390,6 +390,73 @@ internal class TriggersCollector {
       return new TriggerAnnotation(children_contain_killers || expr_is_killer, new_term.Variables, collected_terms);
     }
 
+    private IEnumerable<TriggerTerm> CollectInSetOperations(
+      Expression newExpr, Expression originalExpr, int boundVariables = -1
+    ) {
+      var term = new TriggerTerm { Expr = newExpr, OriginalExpr = originalExpr, Variables = CollectVariables(newExpr) };
+      var newBoundVariables = term.BoundVars.Count();
+      if (newBoundVariables < boundVariables && boundVariables >= 0) {
+        yield break;
+      }
+      yield return term;
+      if (newExpr is SeqSelectExpr
+        {
+          Seq: BinaryExpr
+          {
+            ResolvedOp:
+            BinaryExpr.ResolvedOpcode.MultiSetUnion or
+            BinaryExpr.ResolvedOpcode.MultiSetIntersection or
+            BinaryExpr.ResolvedOpcode.MultiSetDifference,
+            E0: var seq0,
+            E1: var seq1
+          },
+          E0: var seqArg
+        } newSeqSelectExpr) {
+        // Multiset triggers are of the form expression[index]
+        var newTriggerTerm1 = new SeqSelectExpr(newExpr.tok, true, seq0, seqArg, null, newSeqSelectExpr.CloseParen) {
+          Type = Type.Int
+        };
+        foreach (var triggerTerm in CollectInSetOperations(newTriggerTerm1, originalExpr, newBoundVariables)) {
+          yield return triggerTerm;
+        }
+        var newTriggerTerm2 = new SeqSelectExpr(newExpr.tok, true, seq1, seqArg, null, newSeqSelectExpr.CloseParen) {
+          Type = Type.Int
```

Comment:
Shouldn't this type be `seq1.Type`?

Reply:
Good catch, it should actually be `seq1.Type.AsMultiSetType.Arg`